src/omero/plugins/duplicate.py

```diff
@@ -61,6 +72,38 @@ def cmd_type(self):
         import omero.all
         return omero.cmd.Duplicate
 
+    def _pre_objects(self, parser):
+        parser.add_argument(
+            "--duplicate-classes",
+            help=("Modifies the given option by specifying kinds of object to "
+                  "duplicate"))
+        parser.add_argument(
+            "--reference-classes",
```

Comment:
`choices = tuple([x[:-1] for x in dir(omero.model) if x.endswith("I")])` or similar would be able to detect typos in the arguments.

Reply:
Unfortunately that includes classes like `Event` which aren't permitted and omits the example's `Annotation` which is; I don't think we have an easy way here to get a good list of choices.

Reply:
Encoding that logic along with the other base graph methods would probably produce something maintainable:

``` 
def get_graph_targets():
 - load all classes
 - remove events and other known problems
 - add "Annotations" and other necessary base classes
```

Reply:
If not asking the server then probably best done with some codegen adjustments. Marker interfaces like `IAnnotationLink` are also available.

Reply:
Noted in https://github.com/ome/omero-blitz/issues/107.